megaavr/cores/dxcore/low_power.cpp

```diff
@@ -0,0 +1,32 @@
+#include "Arduino.h"
+
+// MALS-89
+void setSleepMode(uint8_t sleepMode) {
+  SLPCTRL.CTRLA = sleepMode; // Bit 0 - SEN - will go back to 0
```

Comment:
Setting the entirety of CTRLA in this manner may have undesired side effects, is there a way we can set the bits of SMODE without affecting any of the other bits in CTRLA?

One example way this could cause a problem is if the user calls enableSleepMode before calling this function, then this function silently disables sleep again by overwriting SEN.

The comment states that SEN will go back to 0 - but this isn't necessarily true. IT depends on the value of sleepMode that is passed. if sleepMode is an odd value, SEN will actually be enabled, which may not be intended.